### PR TITLE
build: update source-map package to latest version

### PR DESCRIPTION
**Pull request by devversion**
See individual commits. Unblocks #46707

### CODE REVIEW

File: modules/playground/e2e_test/example_test.bzl [hunk 1]
```diff
@@ -1,18 +1,25 @@
 load("//tools:defaults.bzl", "protractor_web_test_suite", "ts_library")
 
-def example_test(name, srcs, server, data = [], deps = [], **kwargs):
+def example_test(name, srcs, server, data = [], deps = [], use_legacy_webdriver_types = True, **kwargs):
```
Comment (devversion) [inline modules/playground/e2e_test/example_test.bzl:3]: Keeping this a default with `True` to minimize changes here... we need to clean up this `modules` directory at some point anyway I think.

File: modules/playground/e2e_test/sourcemap/sourcemap_spec.ts [hunk 2]
```diff
@@ -19,9 +19,9 @@ describe('sourcemaps', function() {
 
     $('error-app .errorButton').click();
 
-    browser.manage().logs().get(logging.Type.BROWSER).then(function(logs: any) {
-      let errorLine: number = null;
```
Comment (josephperrott) [inline modules/playground/e2e_test/sourcemap/sourcemap_spec.ts:23]: how did typescript even allow this before?

**Review by josephperrott — APPROVED** [review]
LGTM

File: modules/playground/e2e_test/sourcemap/sourcemap_spec.ts (hunk 2 above)
Comment (devversion) [reply modules/playground/e2e_test/sourcemap/sourcemap_spec.ts:23]: I think the playground tsconfig config does not set `--strict`. We should potentially clean up all of this.. there many small problems in this folder, like tsconfig strictness, the Go-based devserver, benchpress long-term story etc.

**Review by atscott — APPROVED** [review]

**Pull request closed without merging**